consensus/istanbul/core/core.go

```diff
@@ -511,6 +513,19 @@ func (c *core) startNewRound(round *big.Int) error {
 		c.roundChangeSet = newRoundChangeSet(valSet)
 	}
 
+	// Inform the backend that a new sequence has started & bail if the backed stopped the core
+	if !roundChange {
+		if primary := c.backend.IsPrimaryForSeq(newView.Sequence); !primary {
+			// We need to run UpdateReplicaState outside of the core b/c when it stops the core
+			// it runs c.handlerWg.Wait(). To empty that wait group, we need to return from this
+			// function. We unsubscribe from events to stop the core from processing more events
+			// prior to being fully shut down.
+			c.unsubscribeEvents()
+			go c.backend.UpdateReplicaState(newView.Sequence)
```

Comment:
this calls newchainhead which returns an error...
nor updatereplicastate neither the go routine is checking for that error 

doing `go callSomeFuncThaReturnError()` is actually a smell that it's over all our code. We are silencing errors :(